libs/client/WebhookClient.lua

```diff
@@ -0,0 +1,274 @@
+	guildId = "",
+	id = "",
+	token = "",
+	_user = {}
+}
+
+local function parseOptions(customOptions)
+	if type(customOptions) == 'table' then
+		local options = {}
+		for k, default in pairs(defaultOptions) do -- load options
+			local custom = customOptions[k]
+			if custom ~= nil then
+				options[k] = custom
+			else
+				options[k] = default
+			end
+		end
+		for k, v in pairs(customOptions) do -- validate options
+			local default = type(defaultOptions[k])
+			local custom = type(v)
+			if default ~= custom then
+				return error(format('invalid client option %q (%s expected, got %s)', k, default, custom), 3)
+			end
+			if custom == 'number' and (v < 0 or v % 1 ~= 0) then
+				return error(format('invalid client option %q (number must be a positive integer)', k), 3)
+			end
+		end
+		return options
+	else
+		return defaultOptions
+	end
+end
+
+local function resolveImage(avatar, id)
+	if avatar and id then
+		return format("https://cdn.discordapp.com/avatars/%s/%s.png", id, avatar)
+	end
+end
+
+local function parseFile(obj)
+	if type(obj) == 'string' then
+		local data, err = readFileSync(obj)
+		if not data then
+			return nil, err
+		end
+		file = file or {}
+		insert(file, {remove(splitPath(obj)), data})
+	else
+		return nil, 'Invalid file object: ' .. tostring(obj)
+	end
+	return file
+end
+
+
+local WebhookClient, get = require('class')('WebhookClient', Emitter)
+WebhookClient.__description = "Represents a Webhook Client."
+
+function WebhookClient:__init(id, token, options)
+	Emitter.__init(self)
+
+	options = options or {}
+	options["id"] = id
+	options["token"] = token
+
+	self._options = parseOptions(options)
+	self._logger = Logger(self._options.logLevel, self._options.dateTime, self._options.logFile)
+	self._api = API(self)
+
+	-- Sync our defaultOptions with the webhook data
+	local webhook, err = self._api:getWebhookWithToken(self._options.id, self._options.token)
+	if webhook then
+		for key, value in pairs(webhook) do
+			if type(value) ~= "table" then
+				self._options[key] = value
+			end
+		end
+	end
+
+	if options then
+		self:modify(options)
+		for k, v in pairs(options) do
+			self._options[k] = v
+		end
+	end
+end
+
+for name, level in pairs(logLevel) do
+	WebhookClient[name] = function(self, fmt, ...)
+		local msg = self._logger:log(level, fmt, ...)
+		if #self._listeners[name] > 0 then
+			return self:emit(name, msg or format(fmt, ...))
+		end
+	end
+end
+
+-- Getters
+function get.avatar(self)
+	return self._options.avatar
+end
+
+function get.channelId(self)
+	return self._options.channel_id
+end
+
+function get.guildId(self)
+	return self._options.guildId
+end
+
+function get.id(self)
+	return self._options.id
+end
+
+function get.name(self)
+	return self._options.name
+end
+
+function get.token(self)
+	return self._options.token
+end
+
+-- Setters and Modifiers
+function WebhookClient:modify(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
```

Comment:
It's not necessary to check for channel_id, just ignore it like I do with all of the other PATCHes.